internal/actor/lpa_progress_tracker.go

```diff
@@ -0,0 +1,168 @@
+package actor
+
+type ProgressTracker struct {
+	Localizer Localizer
+}
+
+type ProgressTask struct {
+	State TaskState
+	Label string
+}
+
+type Progress struct {
+	Paid                      ProgressTask
+	ConfirmedID               ProgressTask
+	DonorSigned               ProgressTask
+	CertificateProviderSigned ProgressTask
+	AttorneysSigned           ProgressTask
+	LpaSubmitted              ProgressTask
+	StatutoryWaitingPeriod    ProgressTask
+	LpaRegistered             ProgressTask
+}
+
+func (pt ProgressTracker) Progress(donor *DonorProvidedDetails, certificateProvider *CertificateProviderProvidedDetails, attorneys []*AttorneyProvidedDetails) Progress {
+	var (
+		paidLabel,
+		confirmedIDLabel,
+		donorSignedLabel,
+		certificateProviderSignedLabel,
+		attorneysSignedLabel,
+		lpaSubmittedLabel,
+		statutoryWaitingPeriodLabel,
+		lpaRegisteredLabel string
+	)
+
+	if donor.CertificateProvider.FirstNames != "" {
+		certificateProviderSignedLabel = pt.Localizer.Format(
+			"certificateProviderHasDeclared",
+			map[string]interface{}{"CertificateProviderFullName": donor.CertificateProvider.FullName()},
+		)
+	} else {
+		if donor.IsOrganisationDonor() {
+			certificateProviderSignedLabel = pt.Localizer.T("theCertificateProviderHasDeclared")
```

Comment:
I think it might be cleaner to have it structured like this (or similar)

```go
labels := map[string]string{
  "certificateProviderSigned": "yourCertificateProviderHasDeclared",
  // ...
}

if donor.IsOrganisationDonor() {
  labels = map[string]string{
    "certificateProviderSigned": "theCertificateProviderHasDeclared",
    // ...
  }
}

pt.Localizer.T(labels["certificateProviderSigned"])
```

Reply:
although in the figma this is personalised as the certificate provider's name

Reply:
which looks weird, I think it would be nicer to not personalise any of these and have it consistently "the donor", "the certificate provider", etc.